velox/functions/prestosql/tests/ArrayNGramsTest.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <optional>
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox::test;
+
+namespace facebook::velox::functions::test {
+namespace {
+class ArrayNGramsTest : public FunctionBaseTest {
+ protected:
+  template <typename T>
+  void testNgram(
+      const std::vector<std::optional<T>>& inputArray,
+      int64_t n,
+      const std::vector<std::optional<std::vector<std::optional<T>>>>&
+          expectedOutput) {
+    std::vector<std::optional<std::vector<std::optional<T>>>> inputVec(
+        {inputArray});
+    auto input = makeNullableArrayVector<T>(inputVec);
+    auto result =
+        evaluate(fmt::format("ngrams(c0, {})", n), makeRowVector({input}));
+
+    auto expected = makeNullableNestedArrayVector<T>({expectedOutput});
+    assertEqualVectors(expected, result);
+  }
+};
+
+TEST_F(ArrayNGramsTest, integers) {
+  testNgram<int64_t>({1, 2, 3, 4}, 1, {{{1}}, {{2}}, {{3}}, {{4}}});
+  testNgram<int64_t>({1, 2, 3, 4}, 2, {{{1, 2}}, {{2, 3}}, {{3, 4}}});
+  testNgram<int64_t>({1, 2, 3, 4}, 3, {{{1, 2, 3}}, {{2, 3, 4}}});
+  testNgram<int64_t>({1, 2, 3, 4}, 4, {{{1, 2, 3, 4}}});
+  testNgram<int64_t>({1, 2, 3, 4}, 5, {{{1, 2, 3, 4}}});
+  testNgram<int64_t>(
+      {1, 2, 3, 4}, std::numeric_limits<int32_t>::max(), {{{1, 2, 3, 4}}});
```

Comment:
introduce kMax constant for std::numeric_limits<int32_t>::max() to make code easier to read